e2e/test-component/scenarios/embedding-sdk/interactive-question-theming.cy.spec.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+  InteractiveQuestion,
+  type MetabaseTheme,
+} from "@metabase/embedding-sdk-react";
+
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import { createQuestion, describeEE } from "e2e/support/helpers";
+import {
+  mockAuthProviderAndJwtSignIn,
+  mountSdkContent,
+  signInAsAdminAndEnableEmbeddingSdk,
+} from "e2e/support/helpers/component-testing-sdk";
+import { getSdkRoot } from "e2e/support/helpers/e2e-embedding-sdk-helpers";
+import { Box } from "metabase/ui";
+
+const { ORDERS, ORDERS_ID } = SAMPLE_DATABASE;
+
+describeEE(
+  "scenarios > embedding-sdk > interactive-question > theming",
+  // realHover color check is flaky 10% of the time so a retry is added
+  { retries: { runMode: 2, openMode: 2 } },
+  () => {
+    beforeEach(() => {
+      signInAsAdminAndEnableEmbeddingSdk();
+
+      createQuestion({
+        name: "47563",
+        query: {
+          "source-table": ORDERS_ID,
+          aggregation: [["max", ["field", ORDERS.QUANTITY, null]]],
+          breakout: [["field", ORDERS.PRODUCT_ID, null]],
+          limit: 2,
+        },
+      }).then(({ body: question }) => {
+        cy.wrap(question.id).as("questionId");
+      });
+
+      cy.signOut();
+
+      mockAuthProviderAndJwtSignIn();
+    });
+
+    it("derives dynamic css variables for dark theme", () => {
+      const theme: MetabaseTheme = {
+        colors: {
+          background: "rgb(22, 26, 29)",
+          "background-hover": "rgb(14, 17, 20)",
+          "background-disabled": "rgb(45, 45, 48)",
+          "text-primary": "rgb(255, 255, 255)",
+          brand: "rgb(253, 121, 168)",
+        },
+      };
+
+      cy.get<number>("@questionId").then(questionId => {
+        mountSdkContent(
+          <Box bg={theme.colors?.background} h="100vh">
+            <InteractiveQuestion questionId={questionId} />
+          </Box>,
+          { theme },
+        );
+      });
+
+      getSdkRoot().within(() => {
+        cy.findByText("Product ID").should("be.visible");
+
+        const buttonHoverBg = "rgb(33, 39, 43)";
+        const customColumn = "[aria-label='Custom column']";
+
+        // Should be the lightened version of the background color
+        cy.findByTestId("notebook-button")
+          .should("be.visible")
+          .realHover()
+          .should("have.css", "background-color", buttonHoverBg);
+
+        // Should be the lightened version of the background color
+        cy.findByTestId("chart-type-selector-button")
+          .should("be.visible")
+          .realHover()
+          .should("have.css", "background-color", buttonHoverBg);
```

Comment:
The toolbar buttons' hover uses a dynamically defined CSS variable, so we test for those.